server/src/main/java/com/pepperkick/ems/server/service/DesignationService.java

```diff
@@ -0,0 +1,193 @@
+package com.pepperkick.ems.server.service;
+
+import com.pepperkick.ems.server.repository.DesignationRepository;
+import com.pepperkick.ems.server.repository.EmployeeRepository;
+import com.pepperkick.ems.server.requestbody.DesignationRequestPostBody;
+import com.pepperkick.ems.server.entity.Designation;
+import com.pepperkick.ems.server.entity.Employee;
+import com.pepperkick.ems.server.exception.BadRequestException;
+import com.pepperkick.ems.server.exception.NotFoundException;
+import com.pepperkick.ems.server.util.MessageHelper;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.dao.DataIntegrityViolationException;
+import org.springframework.stereotype.Service;
+
+import java.util.List;
+
+@Service
+public class DesignationService {
+    private final DesignationRepository designationRepository;
+    private final EmployeeRepository employeeRepository;
+    private final MessageHelper messageHelper;
+
+    @Autowired
+    public DesignationService(DesignationRepository designationRepository, EmployeeRepository employeeRepository, MessageHelper messageHelper) {
+        this.designationRepository = designationRepository;
+        this.employeeRepository = employeeRepository;
+        this.messageHelper = messageHelper;
+
+        // Check if designation table is empty, if yes then fill with initial data
+        List<Designation> designations = designationRepository.findAll();
+        if (designations.isEmpty()) {
+            String[] titles = { "Director", "Manager", "Lead", "Developer", "DevOps", "QA", "Intern" };
```

Comment:
it would have been nice if you externalised this by reading from a external file or properties or env.